server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -487,24 +487,7 @@ private Filter toFilter(List<String> filterList) {
     }
 
     public JobConfig toJobConfig(CRJob crJob) {
-        JobConfig jobConfig = new JobConfig(crJob.getName());
-        if (crJob.getEnvironmentVariables() != null)
-            for (CREnvironmentVariable crEnvironmentVariable : crJob.getEnvironmentVariables()) {
-                jobConfig.getVariables().add(toEnvironmentVariableConfig(crEnvironmentVariable));
-            }
-
-        List<CRTask> crTasks = crJob.getTasks();
-        Tasks tasks = jobConfig.getTasks();
-        if (crTasks != null)
-            for (CRTask crTask : crTasks) {
-                tasks.add(toAbstractTask(crTask));
-            }
-
-        Tabs tabs = jobConfig.getTabs();
-        if (crJob.getTabs() != null)
-            for (CRTab crTab : crJob.getTabs()) {
-                tabs.add(toTab(crTab));
-            }
+        JobConfig jobConfig = getJobConfig(crJob);
```

Comment:
The method name is misleading. `getJobConfig` seems inappropriate. `createJobConfigFrom(CRJob)` seems more appropriate.

Additionally, it's not clear why the rest of the code was not refactored.